File: rs2-rent-sistem/rs2-ui/rs2_rent_sistem/lib/models/review/review.dart
import 'package:freezed_annotation/freezed_annotation.dart';
import 'package:rs2_rent_sistem/models/item_in_order/order_item.dart';

part 'review.freezed.dart';
part 'review.g.dart';

@Freezed()
class Review with _$Review {
  const factory Review({
    required int id,
    required ItemInOrder orderItem,
    @Default('') String description,
    DateTime? dateAdded,
    required double numberOfStars,
  }) = _Review;

  factory Review.fromJson(Map<String, dynamic> json) => _$ReviewFromJson(json);
}

File: rs2-rent-sistem/rs2-rent-sistem.Model/Models/Review.cs
namespace rs2_rent_sistem.Model.Models
{
    public class Review
    {
        public int ID { get; set; }

        public DateTime? DateAdded { get; set; }

        public string? Description { get; set; }

        public decimal? NumberOfStars { get; set; }
        public int OrderItemID { get; set; }
        public virtual OrderItem OrderItem { get; set; }
    }
}
